src/main/java/slimeknights/tconstruct/gadgets/item/slimesling/PurpleSlimeSlingItem.java

```diff
@@ -0,0 +1,51 @@
+package slimeknights.tconstruct.gadgets.item.slimesling;
+
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.particles.ParticleTypes;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.util.math.vector.Vector3d;
+import net.minecraft.world.World;
+import slimeknights.tconstruct.common.Sounds;
+import slimeknights.tconstruct.library.network.TinkerNetwork;
+import slimeknights.tconstruct.tools.common.network.EntityMovementChangePacket;
+
+public class PurpleSlimeSlingItem extends BaseSlimeSlingItem {
+
+  public PurpleSlimeSlingItem(Properties props) {
+    super(props);
+  }
+
+  /** Called when the player stops using an Item (stops holding the right mouse button). */
+  @Override
+  public void onPlayerStoppedUsing(ItemStack stack, World worldIn, LivingEntity entityLiving, int timeLeft) {
+    if (!(entityLiving instanceof PlayerEntity)) {
+      return;
+    }
+
+    PlayerEntity player = (PlayerEntity) entityLiving;
+    float f = getForce(stack, timeLeft);
+
+    Vector3d look = player.getLookVec();
+    double offX = look.x * f;
+    double offY = look.y * f + 1;
+    double offZ = look.z * f;
```

Comment:
This is a bit hard to use precisely, lets half the power when you are sneaking so if you want to teleport to the other side of a wall you can, otherwise if you want to go faster you just don't crouch. This might be fixing the wrong problem though, address my later comment in this class first and see how it feels.